### ISSUE TITLE
AI Module Prompt for Symptom Checker Logic

### ISSUE THREAD

**Issue by whitehathackerpr** [whitehathackerpr/uganda-telemedicine-app#9]
Design the logic for an AI-based Symptom Checker module that integrates with a Flask backend. The module should load a pre-trained TensorFlow model, accept symptom severity inputs, make predictions, and return the predicted condition and confidence score.

Functional Requirements:
Inputs:

Accept a list of 10 numeric features representing symptom severity (e.g., [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]).

Processing:

Validate the input to ensure it contains exactly 10 numeric values.

Normalize the input data to match the scale used during model training.

Model Integration:

Load a pre-trained TensorFlow model (e.g., saved in .h5 or SavedModel format).

Use the model to make a prediction based on the input features.

Output:

Return a JSON object with the following:

predicted_condition (string): The condition predicted by the model.

confidence_score (float): The confidence level of the prediction.

Implementation Steps:
Model Preparation:

Save the trained TensorFlow model in a suitable format (e.g., .h5).

Ensure the model is compatible with TensorFlow 2.x.

Input Validation:

Check if the input is a valid list of 10 numeric values.

If invalid, return an error response with a message.

Normalization:

Normalize the input features using the same method applied during training (e.g., min-max scaling or standardization).

Prediction Logic:

Load the model once at application startup for efficiency.

Use the predict method to obtain the model's output.

Map the raw prediction output to a human-readable condition label (e.g., ["Condition A", "Condition B"]).

Error Handling:

Handle exceptions during model loading or prediction gracefully.

Output Code Requirements:
Provide Python code for:

Loading the model.

Validating and normalizing inputs.

Making predictions and returning the output in JSON format.

Ensure the module is production-ready and adheres to best practices for performance and security.

Sample AI Module Code:
python
Copy
Edit
import numpy as np
import tensorflow as tf
from flask import Flask, request, jsonify

app = Flask(__name__)

# Load pre-trained TensorFlow model
model = tf.keras.models.load_model("path_to_model/symptom_checker_model.h5")

# Define condition labels
CONDITION_LABELS = ["Condition A", "Condition B", "Condition C", "Condition D"]

# Normalize input features
def normalize_input(features):
    # Example normalization: min-max scaling
    min_val, max_val = 0, 10
    return [(x - min_val) / (max_val - min_val) for x in features]

# Prediction endpoint
@app.route("/api/symptom-checker", methods=["POST"])
def symptom_checker():
    try:
        # Parse JSON input
        data = request.json
        features = data.get("features")

        # Validate input
        if not features or len(features) != 10 or not all(isinstance(f, (int, float)) for f in features):
            return jsonify({"error": "Invalid input. Provide a list of 10 numeric features."}), 400

        # Normalize the input
        normalized_features = normalize_input(features)

        # Convert to numpy array and reshape for the model
        input_array = np.array([normalized_features])

        # Make prediction
        predictions = model.predict(input_array)
        predicted_index = np.argmax(predictions)
        confidence_score = predictions[0][predicted_index]

        # Map to condition label
        predicted_condition = CONDITION_LABELS[predicted_index]

        # Return result
        return jsonify({
            "predicted_condition": predicted_condition,
            "confidence_score": float(confidence_score)
        })

    except Exception as e:
        return jsonify({"error": f"An error occurred: {str(e)}"}), 500

# Health-check endpoint
@app.route("/api/health-check", methods=["GET"])
def health_check():
    return jsonify({"message": "AI Module is running!"})

if __name__ == "__main__":
    app.run(debug=True)
Output Requirements:
The module should handle different levels of symptom severity effectively.

Provide clear error messages for invalid inputs.

Optimize the TensorFlow model loading and prediction process for low latency.

Ensure the code integrates seamlessly with the Flask backend and PostgreSQL database.
